include/SingleCellProtocols.hpp

```diff
@@ -220,7 +221,49 @@ namespace alevin{
       BarcodeEnd end;
     };
 
+    // Custom geometry specification using regex for extraction
+    struct CustomGeo {
+      // store regex string for reads 1 and 2
+      static std::string reg[2];
+      // store positions of matches for bc and umi
+      static std::vector<int> b[2], u[2];
+      // bioRead stores the read number for biological read and bioPat stores match pattern number on regex
+      static unsigned int bioRead, bioPat; // biological read would be contigous and on only 1 of the read
+      static uint32_t minBcLen, maxBcLen, minUmiLen, maxUmiLen;
+      static bool bioReadFound;
+      std::string paddingBases = "ACGT";
+      unsigned int padLen = paddingBases.length();
```

Comment:
Same here, if this doesn't change it could be `constexpr`.  Also, I'd generally prefer a fixed size integral type (e.g. `uint32_t`) to a generic `unsigned int`.